### PR TITLE
Added 'Data Visualization' plugins to create heat maps and icons

### PR DESCRIPTION
**Pull request by Ibrahim5aad**
**Summary:**

This PR introduces two new data visualization plugins to the XbimWebUI: heatmaps and icons. These plugins enhance the visualization capabilities and supporting different data representation scenarios.

**Changes:**

- Heatmaps Plugin: Enables the creation of continuous and discrete color visualizations.
- Icons Plugin: Enables adding icon overlays for IFC elements/products.

**Example:**

A demo example is already added. Simply run the project `npm run start` and choose the 'Data Visualization' example.

### CODE REVIEW

File: examples/data-visualization/index.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+viewer.addPlugin(cube);
+viewer.addPlugin(heatmap);
+viewer.addPlugin(icons);
+
+const tempChannelId = "room_temp";
+const humidityChannelId = "room_humidity";
+
+const temperatureSource = new HeatmapSource("Temp sensor", 1, 152, tempChannelId, 1);
+const humiditySource = new HeatmapSource("Humidity sensor", 1, 152, humidityChannelId, 10);
+const sourceIcon = new Icon("Room 1 Sensor", "Temprature sensor", 1, 152, IconsData.errorIcon);
```
Comment (andyward) [inline examples/data-visualization/index.ts:25]: Typo: Temprature => Temperature
(several others below)

File: examples/data-visualization/index.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+let selectedChannel: ContinuousHeatmapChannel;
+const tempChannel = new ContinuousHeatmapChannel
+(tempChannelId, "double", "Temprature", "Temperature of Rooms", "temprature", "°F", 0, 150, ["#142ce2","#fff200","#FF0000"]);
```
Comment (andyward) [inline examples/data-visualization/index.ts:29]: Let's make this °C and use 0-40 as the range (and update the Random generator below)

File: src/plugins/DataVisualization/Heatmap/heatmap-source.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+export class HeatmapSource {
+    private _id: string;
+    private _modelId: number;
+    private _productId: number;
+    private _channelId: string;
+    private _value: any;
+
+    constructor(id: string, modelId: number, productId: number, channelId: string, value: any) {
```
Comment (andyward) [inline src/plugins/DataVisualization/Heatmap/heatmap-source.ts:8]: Would benefit from some docs on the class and ctor

File: src/plugins/DataVisualization/Heatmap/heatmap-channel.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export interface IHeatmapChannel {
```
Comment (andyward) [inline src/plugins/DataVisualization/Heatmap/heatmap-channel.ts:1]: I'd add some docs - if just to help understand name/description/property etc

File: examples/data-visualization/index.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+const humidityChannel = new ContinuousHeatmapChannel
+(humidityChannelId, "double", "Humidity", "Humidity of Rooms", "temprature", "%", 0, 100, ["#1ac603", "#f96c00"]);
```
Comment (andyward) [inline examples/data-visualization/index.ts:31]: Copy/paste error? 'temprature' should be 'humidity'?

File: examples/data-visualization/index.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
+import { Viewer, NavigationCube, Heatmap, ContinuousHeatmapChannel, HeatmapSource, Icons, CameraType, ViewType, IHeatmapChannel, ClippingPlane, ProductType, } from '../..';
+import { Icon } from '../../src/plugins/DataVisualization/Icons/icon';
+import { Grid } from '../../src/plugins/Grid/grid';
+import { IconsData } from './icons';
+
+const viewer = new Viewer("viewer");
+const cube = new NavigationCube();
```
Comment (andyward) [inline examples/data-visualization/index.ts:7]: For simplicity, lets remove the Grid and Cube from the example. Covered elsewhere

File: examples/data-visualization/index.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+        viewer.camera = CameraType.PERSPECTIVE;
+        clipBox();
+        viewer.resetState(ProductType.IFCSPACE)
+        viewer.show(ViewType.DEFAULT);
+
+        heatmap.addSource(temperatureSource);
+        heatmap.addSource(humiditySource);
+
+        icons.addIcon(sourceIcon);
+        icons.addIcon(new Icon("Temp. Sensor 2", "Temprature sensor", 1, 617, IconsData.successIcon));
```
Comment (andyward) [inline examples/data-visualization/index.ts:49]: Would be nice to demo the onSelected interaction functionality of an icon.

**Review by andyward — COMMENTED** [review]
Looks great. A few bits of feedback @Ibrahim5aad

**Comment by Ibrahim5aad** [comment]
@andyward pushed some commits that addressed your feedback.

**Review by andyward — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**